makefiles/dependency_resolution.inc.mk

```diff
@@ -5,6 +5,9 @@
 OLD_USEMODULE := $(sort $(USEMODULE))
 OLD_USEPKG:= $(sort $(USEPKG))
 
+# add default modules
+USEMODULE += $(filter-out $(DISABLE_MODULE),$(DEFAULT_MODULE))
```

Comment:
I'm not sure I like this, it can make a mess if in one iteration a `MODULE` is added, and because of a dependency in another iteration its `DISABLED`. I think this should not be part of the recursion.

Reply:
I agree that this is something to avoid. However, currently `DEFAULT_MODULE` is not a static array, but gets extended over time.

One way to solve this would be to add this after the recursion. But `DEFAULT_MODULE` could also be extended by the used simply by exporting an environment variable. And users might have done so to add modules that have dependencies on their on. This is something that worked for some significant time, as previously default modules was parsed before the recursion when it still was a static array.

With `DEFAULT_MODULE` no longer being a fixed static array, accessing it once prior to the recursion no longer works. So we have either add it to the recursion, or access it before *and* after. But doing the very some thing at two different places seems to be pretty confusing.

I'd personally would try to make `DEFAULT_MODULE` a static fixed array again and move this prior to the recursion. But to me this discussion seems to be out of scope of this PR.

So for this PR I'd say either keep it like this, or add default modules twice (before and after the recursion). I assume you'd prefer the second?

Reply:
> So for this PR I'd say either keep it like this, or add default modules twice (before and after the recursion). I assume you'd prefer the second?

Yes I think that would be better.

> But to me this discussion seems to be out of scope of this PR.

Agreed.

Reply:
Resolved